javascript/javascript2/week2/homework/hyf-bay/main.js

```diff
@@ -0,0 +1,94 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+const productsList = document.getElementById("productList");
+console.log(products);
+
+//creates and adds INPUT elements to html document
+const searchPanel = document.getElementById("searchPanel");
+const searchProduct = document.createElement("input");
+searchProduct.setAttribute("type", "text");
+searchProduct.setAttribute("id", "searchProduct");
+searchProduct.setAttribute("placeholder", "Product name");
+searchPanel.appendChild(searchProduct);
+
+const searchPrice = document.createElement("input");
+searchPrice.setAttribute("type", "text");
+searchPrice.setAttribute("id", "searchPrice");
+searchPrice.setAttribute("placeholder", "Product price");
+searchPanel.appendChild(searchPrice);
+
+function renderProducts(products) {
+  products.forEach((product) => {
+    const productInfo = document.createElement("li");
+    productInfo.innerHTML = `
+            <ul class="product-info">
+                <li class="product-name">${product.name}</li>
+                <li>Price: ${product.price}</li>
+                <li>Rating: ${product.rating}</li>
+            </ul>
+        `;
+    productsList.appendChild(productInfo);
+  });
+}
+renderProducts(products);
+
+//searching by product name
+searchProduct.addEventListener(`input`, function (event) {
+  const inputProductName = event.target.value;
+  const filteredProductByName = products.filter((product) =>
+    product.name.toLowerCase().indexOf(inputProductName.toLowerCase() != -1)
+  );
+  renderProducts(filteredProductByName);
+});
+
+//searching by product price
+searchPrice.addEventListener("input", function (event) {
+  const inputProductPrice = event.target.value;
+  const filteredProductByPrice = products.filter(function (product) {
+    return product.price <= parseInt(inputProductPrice);
+  });
+  renderProducts(filteredProductByPrice);
+});
+
+//sorting by name
+function sortProductByName(arr) {
+  arr.sort(function (a, b) {
+    if (a.name < b.name) {
+      return -1;
+    }
+    if (a.name > b.name) {
+      return 1;
+    }
+    return 0;
+  });
+  return renderProducts(arr);
+}
+
+function sortProductByPrice(arr) {
+  arr.sort((a, b) => a.price - b.price);
+  return renderProducts(arr);
+}
+
+function sortProductByRating(arr) {
+  arr.sort((a, b) => b.rating - a.rating);
+  return renderProducts(arr);
+}
+
+console.log(sortProductByRating(products));
+
+const sortingDropdown = document.getElementById("sorting");
+sortingDropdown.onchange = function (event) {
+  let optionChosen = sortingDropdown.value;
+  if ((optionChosen = "byName")) {
+    sortProductByName(products);
+  } else if ((optionChosen = "byPrice")) {
+    sortProductByPrice(products);
+  } else if ((optionChosen = "byRating")) {
+    sortProductByRating(products);
+  }
```

Comment:
To compare two things in "if", you should use "==" instead of "=".

const sortingDropdown = document.getElementById("sorting");
sortingDropdown.onchange = function (event) {
let optionChosen = sortingDropdown.value;
if ((optionChosen == "byName")) {
    sortProductByName(products);
  } else if ((optionChosen == "byPrice")) {
    sortProductByPrice(products);
  } else if ((optionChosen == "byRating")) {
    sortProductByRating(products);
  }
};

Dear Veronica,
If you have any other questions, I  will be happy to help.

Reply:
Dear Elham! 
it's insanely cool!!!! I cannot tell you how much I appreciate your help and the time you spent on reviewing all of this code🤯🤩  I'm so thankful.
P.S. About this embarrassing mistake to use assigning operator instead of equality... I think I was out of this planet😅